### DISCUSSION TITLE
Read access to public object from custom javascript v5.0.11

### DISCUSSION THREAD

**Discussion by aetdr** [globaleaks/globaleaks-whistleblowing-software#4231] [General]
Hi @msmannan00 

We try to emhasize the onion link to tenants and need a way to read the public.node.onionservice value from custom javascript.

The following used to work before migrating from v4.15.4 to v5.0.11 :

```
var scope = angular.element(document.body).injector().get("$rootScope");
var tor_onion = '<span class="onionaddress">' + scope.public.node.onionservice + '</span>';

```
The solutions I could find using Google do not work, as the angular object is undefined.

Could you point me in right direction here? 
How to achieve this with AngularJS?

**Comment by msmannan00** [comment]
so here since we migrated from angular JS 1.8.3 to angular 17 the way of interacting with it has changed 
I am working on a small fix in which you would only need 
var tor_onion = '<span class="onionaddress">' + scope.public.node.onionservice + '</span>';
but scope variable has not been exposed at the moment for you to do it, testing it and would make a commit soon

**Reply by aetdr** [reply]
Thank you @msmannan00

**Reply by aetdr** [reply]
@msmannan00 do you have eta when this will get implemented?
I cannot move forward with updating before this.

**Reply by evilaliv3** [reply]
Thank you @aetdr for your question.

At the moment there is no eta. It is our priority to support every users but of course resources are capped.
Do you have ways to sustain some of the development that you need? Please consider donating if you believe in the project.

**Reply by aetdr** [reply]
Hi @evilaliv3 

Unforunately I am not a developer nor I can fund the project at the moment.
Thoght this was a simple thing to fix, as the whole issue is breaking change and it used to work before.

If the fix is complicated to implement, I guess I'll have to remove the customization.
